phpunit/fonts-library/class-wp-font-family-test.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * Tests for Font Family Utils class
+ *
+ * @package Gutenberg
+ */
+
+class WP_Font_Family_Test extends WP_UnitTestCase {
+
+	const FONT_DATA_1 = array(
+		'name'       => 'Piazzolla',
+		'slug'       => 'piazzolla',
+		'fontFamily' => 'Piazzolla',
+		'fontFace'   => array(
+			array(
+				'fontFamily' => 'Piazzolla',
+				'fontStyle'  => 'italic',
+				'fontWeight' => '400',
+				'src'        => 'http://fonts.gstatic.com/s/piazzolla/v33/N0b72SlTPu5rIkWIZjVgI-TckS03oGpPETyEJ88Rbvi0_TzOzKcQhZqx3gX9BRy5m5M.ttf',
+			),
+		),
+	);
+
+	const FONT_DATA_2 = array(
+		'name'       => 'Arial',
+		'slug'       => 'arial',
+		'fontFamily' => 'Arial',
+	);
+
+	function test_get_data() {
+		$font = new WP_Font_Family( self::FONT_DATA_1 );
+		$this->assertEquals( 'piazzolla', $font->get_data()['slug'] );
+		$this->assertEquals( 1, count( $font->get_data()['fontFace'] ) );
+	}
+
+	function test_get_data_as_json() {
+		$font = new WP_Font_Family( self::FONT_DATA_1 );
+		$this->assertEquals( wp_json_encode( self::FONT_DATA_1 ), $font->get_data_as_json() );
+	}
+
+	function test_has_font_faces() {
+		$font1 = new WP_Font_Family( self::FONT_DATA_1 );
+		$this->assertTrue( $font1->has_font_faces() );
+
+		$font2 = new WP_Font_Family( self::FONT_DATA_2 );
+		$this->assertFalse( $font2->has_font_faces() );
+	}
+
+	function test_install_and_uninstall_google_font() {
+		$font = new WP_Font_Family( self::FONT_DATA_1 );
+		$font->install();
+
+		// Check that the post was created
+		$post = $font->get_font_post();
+		$this->assertInstanceof( 'WP_Post', $post );
+
+		// Check that the post has the correct data
+		$this->assertEquals( 'Piazzolla' , $post->post_title );
+		$this->assertEquals( 'piazzolla' , $post->post_name );
+		$content = json_decode( $post->post_content, true );
+		$this->assertEquals( 'Piazzolla' , $content['fontFamily'] );
+		$this->assertEquals( 'piazzolla' , $content['slug'] );
+		$this->assertEquals( 'Piazzolla' , $content['fontFace'][0]['fontFamily'] );
+		$this->assertEquals( 'italic' , $content['fontFace'][0]['fontStyle'] );
+		$this->assertEquals( '400' , $content['fontFace'][0]['fontWeight'] );
+
+		// Check that the font file src was updated to the local font asset
+		$this->assertTrue( str_ends_with( $content['fontFace'][0]['src'], '/piazzolla_italic_400.ttf' ) );
+
+		// Check that the font file was created
+		$this->assertTrue( file_exists( WP_FONTS_DIR . '/piazzolla_italic_400.ttf' ) );
```

Comment:
```suggestion
		$this->assertFileExists( WP_FONTS_DIR . '/piazzolla_italic_400.ttf' );
```
:1234: Applies elsewhere in the PR via `assertFileExists()` and `assertFileDoesNotExist()`

Reply:
done